src/core/usecases/ExecuteTransactionUseCase.ts

```diff
@@ -0,0 +1,129 @@
+import { UseCase } from '../base/use-case';
+import { TransactionType } from '../domain/type/TransactionType';
+import { TransactionRepository } from '../domain/repository/transaction.repository';
+import { PortfolioValueRepository } from '../domain/repository/portfolio-value.repository';
+import { PortfolioTickerRepository } from '../domain/repository/portfolio-ticker.repository';
+import { DailyBarRepository } from '../domain/repository/daily-bar.repository';
+import { UserPortfolioRepository } from '../domain/repository/user-portfolio.repository';
+import { TickerRepository } from '../domain/repository/ticker.repository';
+import { DailyBarNotFoundError } from '../domain/error/DailyBarNotFoundError';
+import { InsufficientCashError } from '../domain/error/InsufficientCashError';
+import { InsufficientHoldingsError } from '../domain/error/InsufficientHoldingsError';
+import { PortfolioNotFoundError } from '../domain/error/PortfolioNotFoundError';
+import { TickerNotFoundError } from '../domain/error/TickerNotFoundError';
+import { PortfolioValueNotFoundError } from '../domain/error/PortfolioValueNotFoundError';
+import { PortfolioTickerNotFoundError } from '../domain/error/PortfolioTickerNotFoundError';
+
+export type ExecuteTransactionCommand = {
+  portfolioId: string;
+  tickerId: string;
+  type: TransactionType;
+  value: number;
+};
```

Comment:
Okay alors attention, tu achètes un volume et non un montant, le montant tu le determine par la suite en multipliant le volume et le prix actuel.

```suggestion
export type ExecuteTransactionCommand = {
  portfolioId: string;
  tickerId: string;
  type: TransactionType;
  volume: number;
};
```

Reply:
comme expliqué dans mon commentaire je peux remplacer "value" par "amount" ou autre désignation monétaire mais pas par un volume